lib/node/neutrino.js

```diff
@@ -0,0 +1,324 @@
+
+    this.opened = false;
+
+    // SPV flag.
+    this.spv = false;
+    this.neutrino = true;
+
+    // Instantiate block storage.
+    this.blocks = blockstore.create({
+      network: this.network,
+      logger: this.logger,
+      prefix: this.config.prefix,
+      cacheSize: this.config.mb('block-cache-size'),
+      memory: this.memory,
+      spv: this.spv,
+      neutrino: this.neutrino
+    });
+
+    this.chain = new Chain({
+      blocks: this.blocks,
+      network: this.network,
+      logger: this.logger,
+      prefix: this.config.prefix,
+      memory: this.memory,
+      maxFiles: this.config.uint('max-files'),
+      cacheSize: this.config.mb('cache-size'),
+      entryCache: this.config.uint('entry-cache'),
+      forceFlags: this.config.bool('force-flags'),
+      checkpoints: this.config.bool('checkpoints'),
+      bip91: this.config.bool('bip91'),
+      bip148: this.config.bool('bip148'),
+      spv: true,
+      neutrino: this.neutrino
+    });
+
+    this.filterIndexers.set(
+      'BASIC',
+      new FilterIndexer({
+        network: this.network,
+        logger: this.logger,
+        blocks: this.blocks,
+        chain: this.chain,
+        memory: this.config.bool('memory'),
+        prefix: this.config.str('index-prefix', this.config.prefix),
+        filterType: 'BASIC',
+        neutrino: true
+      })
+    );
+
+    this.pool = new Pool({
+      network: this.network,
+      logger: this.logger,
+      chain: this.chain,
+      prefix: this.config.prefix,
+      checkpoints: true,
+      filterIndexers: this.filterIndexers,
+      proxy: this.config.str('proxy'),
+      onion: this.config.bool('onion'),
+      upnp: this.config.bool('upnp'),
+      seeds: this.config.array('seeds'),
+      nodes: this.config.array('nodes'),
+      only: this.config.array('only'),
+      maxOutbound: this.config.uint('max-outbound'),
+      createSocket: this.config.func('create-socket'),
+      memory: this.memory,
+      selfish: true,
+      listen: false,
+      neutrino: this.neutrino,
+      spv: this.spv
+    });
+
+    this.rpc = new RPC(this);
+
+    this.http = new HTTP({
+      network: this.network,
+      logger: this.logger,
+      node: this,
+      prefix: this.config.prefix,
+      ssl: this.config.bool('ssl'),
+      keyFile: this.config.path('ssl-key'),
+      certFile: this.config.path('ssl-cert'),
+      host: this.config.str('http-host'),
+      port: this.config.uint('http-port'),
+      apiKey: this.config.str('api-key'),
+      noAuth: this.config.bool('no-auth'),
+      cors: this.config.bool('cors')
+    });
+
+    this.init();
+  }
+
+  /**
+   * Initialize the node.
+   * @private
+   */
+
+  init() {
+    console.log('Initializing Neutrino Node.');
+    // Bind to errors
+    this.chain.on('error', err => this.error(err));
+    this.pool.on('error', err => this.error(err));
+
+    if (this.http)
+      this.http.on('error', err => this.error(err));
+
+    this.pool.on('tx', (tx) => {
```

Comment:
remove this as we don't receive any transactions

Reply:
Oh yeah